src/robin_dict.jl

```diff
@@ -0,0 +1,682 @@
+import Base: setindex!, sizehint!, empty!, isempty, length,
+             getindex, getkey, haskey, iterate, @propagate_inbounds,
+             pop!, delete!, get, get!, isbitstype, in,
+             isiterable, dict_with_eltype, KeySet, Callable
+
+# the load factor arter which the dictionary `rehash` happens
+const ROBIN_DICT_LOAD_FACTOR = 0.80
+
+"""
+    RobinDict([itr])
+
+`RobinDict{K,V}()` constructs a hash table with keys of type `K` and values of type `V`.
+Keys are compared with [`isequal`](@ref) and hashed with [`hash`](@ref).
+Given a single iterable argument, constructs a [`RobinDict`](@ref) whose key-value pairs
+are taken from 2-tuples `(key,value)` generated by the argument.
+
+# Examples
+```jldoctest
+julia> RobinDict([("A", 1), ("B", 2)])
+RobinDict{String,Int64} with 2 entries:
+  "B" => 2
+  "A" => 1
+```
+
+Alternatively, a sequence of pair arguments may be passed.
+
+```jldoctest
+julia> RobinDict("A"=>1, "B"=>2)
+RobinDict{String,Int64} with 2 entries:
+  "B" => 2
+  "A" => 1
+```
+"""
+mutable struct RobinDict{K,V} <: AbstractDict{K,V}
+    #there is no need to maintain an table_size as an additional variable
+    slots::Array{UInt8,1} # indicator, to be used later on
+    keys::Array{K,1}
+    vals::Array{V,1}
+    dibs::Array{Int,1} # distance to initial bucket - critical for implementation
+    count::Int
+    totalcost::Int
```

Comment:
what is a `totalcost`?

Reply:
Kind of like `age` parameter in `Dict`

Reply:
It just seems unfortunate that this is causing there to be two copies of the insert function

Reply:
I can cut that down to one function. That was a pretty lazy implementation on my part.